platform/api/aws/marketplace.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The Mantle Authors
+// SPDX-License-Identifier: Apache-2.0
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/marketplacecatalog"
+)
+
+// UpdateProduct takes care of publishing the AMI to the AWS Marketplace by updating the existing product
+// version.
+// It takes care of scanning the AMI too.
+func (a *API) UpdateProduct(amiID, accessRoleARN, username, version, productID, instanceType string, dryRun bool) error {
+	// the type is always AmiProduct@1.0.
+	// https://docs.aws.amazon.com/marketplace-catalog/latest/api-reference/ami-products.html
+	t := "AmiProduct@1.0"
+
+	catalog := "AWSMarketplace"
+	changeType := "AddDeliveryOptions"
+
+	// seems to be the only way.
+	details := fmt.Sprintf("{\"Version\":{\"VersionTitle\":\"%s\",\"ReleaseNotes\":\"https://www.flatcar.org/releases/#release-%s\"},\"DeliveryOptions\":[{\"Details\":{\"AmiDeliveryOptionDetails\":{\"AmiSource\":{\"AmiId\":\"%s\",\"AccessRoleArn\":\"%s\",\"UserName\":\"%s\",\"OperatingSystemName\":\"OTHERLINUX\",\"OperatingSystemVersion\":\"%s\"},\"UsageInstructions\":\"You can spin up single instances of Flatcar Container Linux using the instructions here: https://docs.flatcar-linux.org/os/booting-on-ec2/\\n\\nFlatcar Container Linux supports a declarative provisioning using the Config format. Learn more from https://docs.flatcar-linux.org/os/provisioning/\",\"RecommendedInstanceType\":\"%s\",\"SecurityGroups\":[{\"IpProtocol\":\"tcp\",\"FromPort\":22,\"ToPort\":22,\"IpRanges\":[\"0.0.0.0/0\"]}]}}}]}", version, version, amiID, accessRoleARN, username, version, instanceType)
```

Comment:
```suggestion
	details := fmt.Sprintf("{\"Version\":{\"VersionTitle\":\"%s\",\"ReleaseNotes\":\"https://www.flatcar.org/releases/#release-%s\"},\"DeliveryOptions\":[{\"Details\":{\"AmiDeliveryOptionDetails\":{\"AmiSource\":{\"AmiId\":\"%s\",\"AccessRoleArn\":\"%s\",\"UserName\":\"%s\",\"OperatingSystemName\":\"OTHERLINUX\",\"OperatingSystemVersion\":\"%s\"},\"UsageInstructions\":\"You can spin up single instances of Flatcar Container Linux using the instructions here: https://www.flatcar.org/docs/latest/installing/cloud/aws-ec2/\\n\\nFlatcar Container Linux supports a declarative provisioning using the Config format. Learn more from https://www.flatcar.org/docs/latest/provisioning/\",\"RecommendedInstanceType\":\"%s\",\"SecurityGroups\":[{\"IpProtocol\":\"tcp\",\"FromPort\":22,\"ToPort\":22,\"IpRanges\":[\"0.0.0.0/0\"]}]}}}]}", version, version, amiID, accessRoleARN, username, version, instanceType)
```

Reply:
Thanks - the current set of release will have this values but we have redirects so it should be fine.